<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string notr="true">Hangman</string>
  </property>
  <property name="windowOpacity">
   <double>1.000000000000000</double>
  </property>
  <property name="iconSize">
   <size>
    <width>24</width>
    <height>24</height>
   </size>
  </property>
  <property name="toolButtonStyle">
   <enum>Qt::ToolButtonStyle::ToolButtonIconOnly</enum>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QStackedWidget" name="stackedWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>800</width>
      <height>600</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>800</width>
      <height>600</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>800</width>
      <height>600</height>
     </size>
    </property>
    <property name="acceptDrops">
     <bool>true</bool>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="frameShape">
     <enum>QFrame::Shape::NoFrame</enum>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="nick_page">
     <widget class="QPushButton" name="nick_submit_btn">
      <property name="geometry">
       <rect>
        <x>360</x>
        <y>360</y>
        <width>81</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>10</pointsize>
       </font>
      </property>
      <property name="text">
       <string>CONNECT</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="nick_field">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>320</y>
        <width>241</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="title_label">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>170</y>
        <width>291</width>
        <height>71</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Javanese Text</family>
        <pointsize>45</pointsize>
        <italic>false</italic>
        <bold>false</bold>
       </font>
      </property>
      <property name="text">
       <string> Hangman </string>
      </property>
     </widget>
     <widget class="QLabel" name="subtitle_label">
      <property name="geometry">
       <rect>
        <x>330</x>
        <y>230</y>
        <width>151</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Javanese Text</family>
        <pointsize>13</pointsize>
       </font>
      </property>
      <property name="text">
       <string> Multiplayer Game </string>
      </property>
     </widget>
     <widget class="QLineEdit" name="ip_field">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>290</y>
        <width>241</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="ip_label">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>290</y>
        <width>51</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Server IP</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="nick_label">
      <property name="geometry">
       <rect>
        <x>206</x>
        <y>320</y>
        <width>61</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Nickname</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="create_or_join_page">
     <widget class="QGroupBox" name="create_groupbox">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>30</y>
        <width>441</width>
        <height>351</height>
       </rect>
      </property>
      <property name="title">
       <string>Create a Room </string>
      </property>
      <widget class="QLineEdit" name="create_name_field">
       <property name="geometry">
        <rect>
         <x>180</x>
         <y>80</y>
         <width>151</width>
         <height>20</height>
        </rect>
       </property>
       <property name="maxLength">
        <number>20</number>
       </property>
      </widget>
      <widget class="QLineEdit" name="create_password_field">
       <property name="geometry">
        <rect>
         <x>180</x>
         <y>110</y>
         <width>151</width>
         <height>20</height>
        </rect>
       </property>
       <property name="maxLength">
        <number>20</number>
       </property>
      </widget>
      <widget class="QComboBox" name="level_combobox">
       <property name="geometry">
        <rect>
         <x>180</x>
         <y>140</y>
         <width>151</width>
         <height>22</height>
        </rect>
       </property>
      </widget>
      <widget class="QSpinBox" name="rounds_number_spin">
       <property name="geometry">
        <rect>
         <x>180</x>
         <y>170</y>
         <width>151</width>
         <height>22</height>
        </rect>
       </property>
      </widget>
      <widget class="QTimeEdit" name="time_edit">
       <property name="geometry">
        <rect>
         <x>180</x>
         <y>200</y>
         <width>151</width>
         <height>22</height>
        </rect>
       </property>
      </widget>
      <widget class="QPushButton" name="create_btn">
       <property name="geometry">
        <rect>
         <x>210</x>
         <y>240</y>
         <width>91</width>
         <height>23</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>10</pointsize>
        </font>
       </property>
       <property name="text">
        <string>CREATE</string>
       </property>
      </widget>
      <widget class="QLabel" name="name_label">
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>80</y>
         <width>61</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Name</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="pass_label">
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>110</y>
         <width>61</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string> Password</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="level_label">
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>140</y>
         <width>61</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string> Difficulty</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="number_label">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>170</y>
         <width>111</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string> Number of Rounds</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="time_label">
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>200</y>
         <width>91</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string> Time per Round</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="join_groupbox">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>390</y>
        <width>441</width>
        <height>171</height>
       </rect>
      </property>
      <property name="title">
       <string>Join a Room </string>
      </property>
      <widget class="QLineEdit" name="join_password_field">
       <property name="geometry">
        <rect>
         <x>180</x>
         <y>80</y>
         <width>151</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="maxLength">
        <number>20</number>
       </property>
      </widget>
      <widget class="QLineEdit" name="join_room_name_field">
       <property name="geometry">
        <rect>
         <x>180</x>
         <y>50</y>
         <width>151</width>
         <height>20</height>
        </rect>
       </property>
       <property name="layoutDirection">
        <enum>Qt::LayoutDirection::LeftToRight</enum>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="maxLength">
        <number>20</number>
       </property>
      </widget>
      <widget class="QLabel" name="name_2_label">
       <property name="geometry">
        <rect>
         <x>110</x>
         <y>50</y>
         <width>51</width>
         <height>20</height>
        </rect>
       </property>
       <property name="layoutDirection">
        <enum>Qt::LayoutDirection::LeftToRight</enum>
       </property>
       <property name="text">
        <string> Name</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="pass_2_label">
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>80</y>
         <width>61</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string> Password</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
       </property>
      </widget>
      <widget class="QPushButton" name="join_btn">
       <property name="geometry">
        <rect>
         <x>210</x>
         <y>120</y>
         <width>91</width>
         <height>23</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>10</pointsize>
        </font>
       </property>
       <property name="text">
        <string>JOIN</string>
       </property>
      </widget>
     </widget>
     <widget class="QListWidget" name="rooms_list">
      <property name="geometry">
       <rect>
        <x>500</x>
        <y>40</y>
        <width>271</width>
        <height>521</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="list_of_rooms_label">
      <property name="geometry">
       <rect>
        <x>600</x>
        <y>20</y>
        <width>101</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>  List of Rooms </string>
      </property>
     </widget>
     <widget class="QPushButton" name="create_join_back_btn">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>570</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>BACK</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="waitroom_page">
     <widget class="QLabel" name="waiting_label">
      <property name="geometry">
       <rect>
        <x>210</x>
        <y>140</y>
        <width>361</width>
        <height>71</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Javanese Text</family>
        <pointsize>20</pointsize>
       </font>
      </property>
      <property name="text">
       <string>waiting for the game to start...</string>
      </property>
     </widget>
     <widget class="QListWidget" name="players_list">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>200</y>
        <width>371</width>
        <height>171</height>
       </rect>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="MinimumExpanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>21</height>
       </size>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
     </widget>
     <widget class="QPushButton" name="start_btn">
      <property name="geometry">
       <rect>
        <x>340</x>
        <y>390</y>
        <width>101</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>15</pointsize>
       </font>
      </property>
      <property name="text">
       <string>START</string>
      </property>
     </widget>
     <widget class="QPushButton" name="waitroom_back_btn">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>570</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>BACK</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="game_page">
     <widget class="QLabel" name="word_label">
      <property name="geometry">
       <rect>
        <x>450</x>
        <y>170</y>
        <width>231</width>
        <height>61</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Javanese Text</family>
        <pointsize>30</pointsize>
       </font>
      </property>
      <property name="text">
       <string>_ _ _ _ _ _ _</string>
      </property>
     </widget>
     <widget class="QLabel" name="rounds1_label">
      <property name="geometry">
       <rect>
        <x>510</x>
        <y>50</y>
        <width>71</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>ROUNDS</string>
      </property>
     </widget>
     <widget class="QLabel" name="time1_label">
      <property name="geometry">
       <rect>
        <x>510</x>
        <y>80</y>
        <width>51</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>TIME</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QLineEdit" name="letter_input">
      <property name="geometry">
       <rect>
        <x>550</x>
        <y>350</y>
        <width>31</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>16</pointsize>
       </font>
      </property>
      <property name="inputMethodHints">
       <set>Qt::InputMethodHint::ImhNone</set>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="maxLength">
       <number>1</number>
      </property>
      <property name="clearButtonEnabled">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QListWidget" name="ranking_list">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>40</y>
        <width>301</width>
        <height>101</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="rounds2_label">
      <property name="geometry">
       <rect>
        <x>590</x>
        <y>40</y>
        <width>61</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>16</pointsize>
       </font>
      </property>
      <property name="text">
       <string>0/0</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignBottom|Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft</set>
      </property>
     </widget>
     <widget class="QLabel" name="time2_label_2">
      <property name="geometry">
       <rect>
        <x>590</x>
        <y>70</y>
        <width>91</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>16</pointsize>
       </font>
      </property>
      <property name="text">
       <string>0:00</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignBottom|Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft</set>
      </property>
     </widget>
     <widget class="QPushButton" name="leave_room_btn">
      <property name="geometry">
       <rect>
        <x>680</x>
        <y>60</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Leave Room</string>
      </property>
     </widget>
     <widget class="QLabel" name="ranking_label">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>15</y>
        <width>71</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Ranking</string>
      </property>
     </widget>
     <widget class="QLabel" name="player0_label">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>170</y>
        <width>300</width>
        <height>260</height>
       </rect>
      </property>
      <property name="text">
       <string>TextLabel</string>
      </property>
     </widget>
     <widget class="QLabel" name="player1_label">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>450</y>
        <width>150</width>
        <height>130</height>
       </rect>
      </property>
      <property name="text">
       <string>TextLabel</string>
      </property>
     </widget>
     <widget class="QLabel" name="player2_label">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>450</y>
        <width>150</width>
        <height>130</height>
       </rect>
      </property>
      <property name="text">
       <string>TextLabel</string>
      </property>
     </widget>
     <widget class="QLabel" name="player3_label">
      <property name="geometry">
       <rect>
        <x>420</x>
        <y>450</y>
        <width>150</width>
        <height>130</height>
       </rect>
      </property>
      <property name="text">
       <string>TextLabel</string>
      </property>
     </widget>
     <widget class="QLabel" name="player4_label">
      <property name="geometry">
       <rect>
        <x>610</x>
        <y>450</y>
        <width>150</width>
        <height>130</height>
       </rect>
      </property>
      <property name="text">
       <string>TextLabel</string>
      </property>
     </widget>
     <widget class="QLabel" name="fails_label">
      <property name="geometry">
       <rect>
        <x>440</x>
        <y>250</y>
        <width>251</width>
        <height>91</height>
       </rect>
      </property>
      <property name="text">
       <string>TextLabel</string>
      </property>
     </widget>
     <widget class="QPushButton" name="send_letter_btn">
      <property name="geometry">
       <rect>
        <x>590</x>
        <y>355</y>
        <width>21</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>7</pointsize>
       </font>
      </property>
      <property name="text">
       <string>OK</string>
      </property>
      <property name="checkable">
       <bool>false</bool>
      </property>
      <property name="autoDefault">
       <bool>false</bool>
      </property>
      <property name="default">
       <bool>false</bool>
      </property>
      <property name="flat">
       <bool>false</bool>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="end_page">
     <widget class="QListView" name="ranking_list_2">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>180</y>
        <width>321</width>
        <height>211</height>
       </rect>
      </property>
      <property name="editTriggers">
       <set>QAbstractItemView::EditTrigger::NoEditTriggers</set>
      </property>
      <property name="defaultDropAction">
       <enum>Qt::DropAction::IgnoreAction</enum>
      </property>
      <property name="isWrapping" stdset="0">
       <bool>false</bool>
      </property>
      <property name="resizeMode">
       <enum>QListView::ResizeMode::Fixed</enum>
      </property>
      <property name="selectionRectVisible">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QLabel" name="label_10">
      <property name="geometry">
       <rect>
        <x>340</x>
        <y>150</y>
        <width>121</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Javanese Text</family>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Final Ranking</string>
      </property>
     </widget>
     <widget class="QPushButton" name="restart_btn">
      <property name="geometry">
       <rect>
        <x>330</x>
        <y>430</y>
        <width>151</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>11</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Play Again</string>
      </property>
     </widget>
     <widget class="QPushButton" name="end_back_btn">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>570</y>
        <width>75</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>BACK</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
